include/crocoddyl/multibody/actuations/floating-base.hpp

```diff
@@ -54,9 +54,9 @@ class ActuationModelFloatingBaseTpl : public ActuationModelAbstractTpl<_Scalar>
   virtual boost::shared_ptr<ActuationDataAbstract> createData() {
     boost::shared_ptr<ActuationDataAbstract> data = boost::make_shared<ActuationDataAbstract>(this);
     data->dtau_du.diagonal(-6).fill((Scalar)1.);
-
-    assert_pretty(dtau_du_ == data->dtau_du, "dtau_du has wrong value.");
-
+#ifndef NDEBUG
+    dtau_du_ = data->dtau_du;
```

Comment:
What is the purpose of this assignment when `NDEBUG` is active? I.e., how is that variable set when `NDEBUG` is _not_ active?

Reply:
To be used for unit-testing. The previously modified code was producing some fails in the unittest.

Reply:
So, `dtau_du_` is not set otherwise but only set in unit testing? If so, a comment should be added - because that means we have an internal member _solely_ for unit testing. This is confusing indeed.

Reply:
`dtau_du_` exist only for Debug mode.

Reply:
I think it is better if you get rid of this quantity. We don't need this here.

Reply:
This checks is important! If you check twice, `data->dtau_du` is defined once (during the data creation). Then if the user passes a wrong data class, we will have an error here.

With this method, I avoid to fill again and again a matrix that is always constant. But, of course, we need some test here.

Reply:
I think this is a very bad practice to have data that are only manipulated in Debug mode. It will cause some troubles sooner or later. If the purpose is to check the validity of user's mistakes, there might be other but secured solutions.

Reply:
At the moment, I don't see a better way of testing the user validity. Better ideas are welcome!

Due to this issue is orthogonal to the main purpose of the PR and there is no a clear solution now, I propose to work on it later.